src/ui/egg-summary-ui-handler.ts

```diff
@@ -0,0 +1,316 @@
+import BattleScene from "../battle-scene";
+import { Mode } from "./ui";
+import PokemonIconAnimHandler, { PokemonIconAnimMode } from "./pokemon-icon-anim-handler";
+import MessageUiHandler from "./message-ui-handler";
+import { getEggTierForSpecies } from "../data/egg";
+import {Button} from "#enums/buttons";
+import { Gender } from "#app/data/gender.js";
+import { getVariantTint } from "#app/data/variant.js";
+import { EggTier } from "#app/enums/egg-type.js";
+import PokemonHatchInfoContainer from "./pokemon-hatch-info-container";
+import { EggSummaryPhase } from "#app/phases/egg-summary-phase.js";
+import { DexAttr } from "#app/system/game-data.js";
+import { EggHatchData } from "#app/data/egg-hatch-data.js";
```

Comment:
```suggestion
import { Gender } from "#app/data/gender";
import { getVariantTint } from "#app/data/variant";
import { EggTier } from "#app/enums/egg-type";
import PokemonHatchInfoContainer from "./pokemon-hatch-info-container";
import { EggSummaryPhase } from "#app/phases/egg-summary-phase";
import { DexAttr } from "#app/system/game-data";
import { EggHatchData } from "#app/data/egg-hatch-data";
```
NIT: Same comment as above